### ISSUE TITLE
tf2.0, tf.keras Embedding layer behave different than tf1.13

### ISSUE THREAD

**Issue by BrikerMan** [tensorflow/tensorflow#28501]
<em>Please make sure that this is a bug. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:bug_template</em>

**System information**
- OS Platform and Distribution: Ubuntu 16.04
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below): 2.0.0-alpha0
- Python version: 3.6.7

```python
from tensorflow import keras
import numpy as np

L = keras.layers

embedding_matrix = np.random.random((10, 5))

model = keras.Sequential([
    L.Embedding(input_dim=10, 
                output_dim=5,
                weights=[embedding_matrix],
                trainable=False)
])

model.compile('rmsprop', 'mse')

embedding_matrix[2] == model.predict([2])[0][0]
```

**Describe the current behavior**

`embedding_matrix[2]` is not equal to  `model.predict([2])[0][0]`.

**Describe the expected behavior**

`embedding_matrix[2]` should equal to  `model.predict([2])[0][0]`, those two equals on tf 1.13.1 version.

**Comment by ymodak** [comment]
@BrikerMan I was not able to reproduce the behavior reported.  I executed the above code snippet in TF 1.13.1 and 2.0 alpha to receive identical results as given below:
```python
array([False, False, False, False, False])
```
Can you please try executing your code in [google colab](https://colab.sandbox.google.com/notebooks/welcome.ipynb#recent=true) and confirm? Thanks!

**Comment by BrikerMan** [comment]
@ymodak I have tried it with colab, it should be equal when using 1.13.1.

I have changed the last line of my code and it works as expected.

```
model.predict([2])[0][0] == np.array(embedding_matrix[2], dtype=np.float32)
```

result is 
```python
# 1.13.1
array([ True,  True,  True,  True,  True])
# 2.0 alpha
array([False, False, False, False, False])
```

Here is demo colab: https://colab.research.google.com/drive/1Y1ObMAg3H2O47on58t2hdirRUr6M172o

**Comment by robieta** [comment]
Thanks for the report. (And excellent example!)

What's happening is that in eager we don't pick up the initial weights in eager mode. (Which definitely needs to be fixed.) https://github.com/tensorflow/tensorflow/blob/f0836d2a3bdc83b9487d703f30669723bd2662fb/tensorflow/python/keras/engine/base_layer.py#L683

We (the keras team) will triage and report back.

**Comment by ymodak** [comment]
@BrikerMan This appears to be fixed with latest 2.0 nightly (2.0.0-dev20190612) Can you please confirm?

**Comment by BrikerMan** [comment]
> @BrikerMan This appears to be fixed with latest TF 2.0 nightly (2.0.0-dev20190612) Can you please confirm?

Fixed, thanks~
